### PR TITLE
[SYCL] Added test for deprecated features

### PR DESCRIPTION
**Pull request by maximdimakov**
The test is created for the check that ABI of deprecated API is not broken after this API was removed. Removed API was wrapped in __SYCL_INTERNAL_API macro. Also this test checks that removed API is working with this macro.
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
Could you provide more details for the purpose of the test?
It is unclear what is actually tested especially because there is no check condition step.

**Comment by vladimirlaz** [comment]
if a new directory is added please update .github/CODEOWNERS.

**Comment by maximdimakov** [comment]
@vladimirlaz I updated the description of PR.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -71,3 +71,6 @@ SYCL/USM @jbrodman
 
 # Stream
 SYCL/Basic/stream @againull
+
+# Deprecated features
+SYCL/Deprecated_features @maximdimakov @alexbatashev
```
Comment (alexbatashev) [inline .github/CODEOWNERS]:
```suggestion
SYCL/DeprecatedFeatures @intel/llvm-reviewers-runtime
```

**Review by romanovvlad — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**